src/main/java/com/eppo/sdk/EppoClient.java

```diff
@@ -85,25 +87,40 @@ private Optional<EppoValue> getAssignmentValue(
             return Optional.empty();
         }
 
-        // Find matched rule
-        Optional<Rule> rule = RuleValidator.findMatchingRule(subjectAttributes, configuration.getRules());
-        if (rule.isEmpty()) {
-            log.info("[Eppo SDK] No assigned variation. The subject attributes did not match any targeting rules");
-            return Optional.empty();
-        }
-
-        // Check if in experiment sample
-        String allocationKey = rule.get().getAllocationKey();
-        Allocation allocation = configuration.getAllocation(allocationKey);
-        if (!this.isInExperimentSample(subjectKey, flagKey, configuration.getSubjectShards(),
-                allocation.getPercentExposure())) {
-            log.info("[Eppo SDK] No assigned variation. The subject is not part of the sample population");
-            return Optional.empty();
+        String allocationKey;
+        List<Variation> variations;
```

Comment:
To leverage as much existing code as possible, my approach was to have bandit models generate a list of variations that our existing variation code then plugs into.

So, if it's a bandit, we pass all the pertinent information to a `BanditEvaluator`, producing the list of variations. Otherwise, we use rules and allocations like we always have.

Before this code, we shared the logic for checking if it's enabled or if there are any overrides.

Once we have our list of variations, it doesn't matter if it's a bandit or not (for now; we'll want to extend the logging later as well)